borg/testsuite/repository.py

```diff
@@ -216,18 +216,18 @@ def segments_in_repository():
         self.repository.commit()
 
         self.repository.append_only = False
-        assert segments_in_repository() == 1
+        assert segments_in_repository() == 2
         self.repository.put(b'00000000000000000000000000000000', b'foo')
         self.repository.commit()
         # normal: compact squashes the data together, only one segment
-        assert segments_in_repository() == 1
+        assert segments_in_repository() == 4
```

Comment:
The log was the same before, just distributed differently (all in one segment). In this instance it looks really wasteful, but for transactions containing data it's not a big difference. E.g. `<couple hundred MB> <commit> <couple hundred MB> <commit> <some data copied from sparse segments> <commit>`